llama-index-integrations/retrievers/llama-index-retrievers-vertexai-search/llama_index/retrievers/vertexai_search/base.py

```diff
@@ -0,0 +1,389 @@
+        note_with_score: List[NodeWithScore] = []
+
+        for result in results:
+            document_dict = MessageToDict(
+                result.document._pb, preserving_proto_field_name=True
+            )
+            derived_struct_data = document_dict.get("derived_struct_data")
+            if not derived_struct_data:
+                continue
+
+            if chunk_type not in derived_struct_data:
+                continue
+
+            for chunk in derived_struct_data[chunk_type]:
+                score = chunk.get("relevanceScore", 0)
+                note_with_score.append(
+                    NodeWithScore(
+                        node=TextNode(text=chunk.get("content", "")),
+                        score=score,
+                    )
+                )
+
+        return note_with_score
+
+    def _convert_website_datastore_response(
+        self, results: Sequence[SearchResult], chunk_type: str
+    ) -> List[NodeWithScore]:
+        """Converts a sequence of search results to a list of LLamaindex note_with_score."""
+        note_with_score: List[NodeWithScore] = []
+
+        for result in results:
+            document_dict = MessageToDict(
+                result.document._pb, preserving_proto_field_name=True
+            )
+
+            derived_struct_data = document_dict.get("derived_struct_data")
+            if not derived_struct_data:
+                continue
+
+            if chunk_type not in derived_struct_data:
+                continue
+
+            text_field = "snippet" if chunk_type == "snippets" else "content"
+
+            for chunk in derived_struct_data[chunk_type]:
+                score = chunk.get("relevanceScore", 0)
+                note_with_score.append(
+                    NodeWithScore(
+                        node=TextNode(text=chunk.get(text_field, "")),
+                        score=score,
+                    )
+                )
+
+        if not note_with_score:
+            print(f"No {chunk_type} could be found.")
+            if chunk_type == "extractive_answers":
+                print(
+                    "Make sure that your data store is using Advanced Website "
+                    "Indexing.\n"
+                    "https://cloud.google.com/generative-ai-app-builder/docs/about-advanced-features#advanced-website-indexing"
+                )
+
+        return note_with_score
+
+    def _retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
+        """Retrieve from the platform."""
+        """Get note_with_score relevant for a query."""
+
+        search_request = self._create_search_request(query_bundle.query_str)
+
+        try:
+            response = self._client.search(search_request)
+        except InvalidArgument as exc:
+            raise type(exc)(
+                exc.message
+                + " This might be due to engine_data_type not set correctly."
+            )
+
+        if self.engine_data_type == 0:
```

Comment:
Its a little weird to me that this is an int instead of string, but I guess that probably aligns with some google setting somewhere I hope?

Reply:
Thanks. Good question. It could be string. We just follow the similar way as we implemented for other frameworks.
There're different data stores: structured, unstructured, website based, etc. This variable is used to identify the data store type to use.